tests/test_main.py

```diff
@@ -87,39 +87,54 @@ def test_prog():
 
 
 @pytest.mark.isolated
-def test_build(mocker, test_flit_path):
```

Comment:
This test actually contained five different tests, and because was causing me some issues I've split it up into five separate tests within this PR.